spec/zipkin/scope_manager_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+
+RSpec.describe Zipkin::ScopeManager do
+  let(:scope_manager) { described_class.new }
+  let(:span) { instance_spy(Zipkin::Span) }
+
+  it 'activates a span' do
+    scope_manager.activate(span)
+    expect(scope_manager.active.span).to eq(span)
+  end
+
+  it 'can change the active span' do
+    span2 = instance_spy(Zipkin::Span)
+
+    scope_manager.activate(span)
+    scope_manager.activate(span2)
+    expect(scope_manager.active.span).to eq(span2)
+  end
+
+  it 'changes active span back to the previous active span on active scope close' do
```

Comment:
Readability might benefit from breaking out context (closing active span) and behavior / post-condition (active span reverted to previous).